### PR TITLE
Multistage render

### PR DESCRIPTION
**Pull request by flurin**
The request to have multi-stage rendering with Tilt (i.e. that you have a .md.erb file that will first process erb and then markdown) made me re-envision the way rendering works in Roger.

This also opened up the way to simplify some of the code in the helpers. There is now only one render method "to rule them all" and it's called `Render#render`.

This also introduces a new helper `render_file` to enable you to render any template on disk.

### CODE REVIEW

File: lib/roger/template/helpers/capture.rb [hunk 1]
```diff
@@ -27,8 +27,10 @@ def content_for(block_name, &block)
 
         # rubocop:disable Lint/Eval
         def capture(&block)
-          unless template.template.is_a?(Tilt::ERBTemplate)
-            fail ArgumentError, "content_for works only with ERB Templates"
+          unless template.current_tilt_template == Tilt::ERBTemplate
+            err  = "content_for works only with ERB Templates"
```
Comment (flurin) [inline lib/roger/template/helpers/capture.rb]: Lines up with line below

**Comment by edwinvdgraaf** [comment]
Just gave this a spin, and works like a charm so far.


Yet .. right now my file is named: `index2.html.md` and is resolved under the url `http://localhost:9000/index2.html.md`. But perhaps this has more to do with the resolver?

**Comment by flurin** [comment]
What did you expect it to be?

**Comment by flurin** [comment]
This may indeed an issue with the resolver; But I don't know how and if we can fix that. The triple extensions shouldn't be necessary btw. A simple `index.md.erb` should suffice to translate to `index.html` 

If what I write above does not work, can you make a list of expectations? Just so we know what we would want?

**Comment by edwinvdgraaf** [comment]
> A simple index.md.erb should suffice to translate to index.html
This doesn't work.

>  a list of expectations
Basically this depends on input type: does `tilt` tracks this?

**Comment by flurin** [comment]
Nope. Unfortunately it doesn't. 

Also the problem is that we have to search for the real file based on an url. So to understand the situation better I was looking for some examples of what we'd want. For instance:

index.html -> index.html.erb
index.html -> index.md.erb
index.html -> index.md
index.html -> index.html

**Pull request merged**